coderd/notifications/notifier.go

```diff
@@ -223,6 +229,18 @@ func (n *notifier) prepare(ctx context.Context, msg database.AcquireNotification
 		return nil, xerrors.Errorf("failed to resolve handler %q", msg.Method)
 	}
 
+	logoURL, err := n.store.GetLogoURL(ctx)
+	if err != nil && !errors.Is(err, sql.ErrNoRows) {
+		n.log.Error(ctx, "failed fetching logo url", slog.Error(err))
+	}
+
+	if logoURL == "" {
+		//nolint:ineffassign // define to default value if unable to fetch one from db
+		logoURL = NotificationsDefaultLogoURL
+	}
+
+	payload.Labels["_logo_url"] = logoURL
```

Comment:
I agree with @mtojek here.

I don't think this should go into the payload. It's going to be the same for every notification, and if it's in the payload then we'll be storing the same URL for every notification.

Have a look at `coderd/notifications/dispatch/smtp/html.gotmpl: you'll see `{{ base_url }}`.
This is provided by a helper function, defined in `cli/server.go`:

```go
// templateHelpers builds a set of functions which can be called in templates.
// We build them here to avoid an import cycle by using coderd.Options in notifications.Manager.
// We can later use this to inject whitelabel fields when app name / logo URL are overridden. <----
func templateHelpers(options *coderd.Options) map[string]any {
	return map[string]any{
		"base_url":     func() string { return options.AccessURL.String() },
		"current_year": func() string { return strconv.Itoa(time.Now().Year()) },
	}
}
```

We should create a new helper which retrieves the logo instead, I think.